omx/gstomxvideodec.c

```diff
@@ -2655,7 +2657,7 @@ gst_omx_video_dec_drain (GstVideoDecoder * decoder, gboolean is_eos)
 
   klass = GST_OMX_VIDEO_DEC_GET_CLASS (self);
 
-  if (!self->started) {
+  if (!self->started || self->flush_flag) {
```

Comment:
protect flush_flag and started

Reply:
The thread is already locked, it gets unlocked in line 2683, and locked again when this function returns.